client/src/Ratings.jsx

```diff
@@ -1,59 +1,65 @@
+/* eslint-disable import/extensions */
 import React, { Component } from 'react';
-import PropTypes from 'prop-types';
+import axios from 'axios';
+
 import ReviewAction from './RatingsComponent/ReviewAction.jsx';
 import RatingsSearch from './RatingsComponent/RatingsSearch.jsx';
 import RatingsContent from './RatingsComponent/RatingsContent.jsx';
 import RatingsStarHeader from './RatingsComponent/RatingsStarHeader.jsx';
 import RatingsSummaryReview from './RatingsComponent/RatingsSummaryReview.jsx';
-const { data } = require('./RatingsComponent/RatingsDummyData.js');
-
-let newData = data.slice(0,2);
 
 class Ratings extends Component {
-  constructor(props) {
-    super(props);
+  constructor() {
+    super();
     this.state = {
-      ratings: newData,
+      ratingsStorage: [],
+      limitedRatings: [],
       ratingsCount: 2,
       showMoreRatings: true,
     };
     this.onAddMoreReviews = this.onAddMoreReviews.bind(this);
   }
 
-  // componentDidMount() {
-  //   axios.get('/api/reviews/?product_id=48487')
-  //     .then((results) => {
-  //       this.setState({
-  //         reviews: results.data.results,
-  //       });
-  //     })
-  //     .catch((err) => {
-  //       console.log(err);
-  //     });
-  // }
+  componentDidMount() {
+    this.getProductRatings();
+  }
 
   onAddMoreReviews() {
-    const { ratingsCount } = this.state;
+    const { ratingsStorage, ratingsCount } = this.state;
+    const ratingStorageLen = ratingsStorage.length;
 
     let newRatingsCount = ratingsCount + 2;
     let anyMoreRatings = true;
 
-    if (data.length < newRatingsCount) {
-      newRatingsCount = data.length;
+    if (ratingStorageLen < newRatingsCount) {
+      newRatingsCount = ratingStorageLen;
       anyMoreRatings = false;
     }
 
-    const newListings = data.slice(0, newRatingsCount);
+    const newListings = ratingsStorage.slice(0, newRatingsCount);
 
     this.setState({
-      ratings: newListings,
+      limitedRatings: newListings,
       ratingsCount: newRatingsCount,
       showMoreRatings: anyMoreRatings,
     });
-  };
+  }
+
+  getProductRatings() {
```

Comment:
I think this is missing from the constructor, i.e. it's not bound with "this"

Reply:
Yes you are correct. The two reasons that i did not include binding this method in the constructor is: 1.) the method is not getting passed into any other component therefore I do not have to worry about loosing the binding. 2.) Going to refactor the Ratings.jsx to be a functional component so this will change how this method is being created.